refnx/reduce/platypusnexus.py

```diff
@@ -626,6 +627,166 @@ def _check_sample_environments(self, d, h5d):
+                pass
+            else:
+                try:
+                    channel = os.path.join(data_folder, channel)
+                except TypeError:
+                    # original channel is not a string
+                    pass
+                finally:
+                    # let's hope it's an h5 file
+                    channel = self.reflect_klass(channel)
+
+            if channel.spin_state is SpinChannel.DOWN_DOWN:
+                self.dd = channel
+            elif channel.spin_state is SpinChannel.DOWN_UP:
+                self.du = channel
+            elif channel.spin_state is SpinChannel.UP_DOWN:
+                self.ud = channel
+            elif channel.spin_state is SpinChannel.UP_UP:
+                self.uu = channel
+
+        assert (
+            self.dd is not None
+        ), "down_down spin channel is not SpinChannel.DOWN_DOWN!"
+        assert (
+            self.uu is not None
+        ), "up_up spin channel is not SpinChannel.UP_UP!"
+
+    @property
+    def spin_channels(self):
+        return [
+            s.spin_state.value if s is not None else None
+            for s in [self.dd, self.du, self.ud, self.uu]
+        ]
+
+    def process_beams(self, reduction_options=None):
+        """
+        Process beams in SpinSet.
+
+        Reduction options for each spin channel are specified by
+        self.dd_opts, self.du_opts, self.ud_opts, and self.uu_opts where
+        a standard set of options is provided when constructing the object.
+        To specify different options for each spin channel (such as using
+        the ManualBeamFinder for only spin-flip channels), update the
+        reduction options for the specific spin channel in SpinSet, then
+        process the beams. i.e.
+
+        from refnx.reduce.manual_beam_finder import ManualBeamFinder
+        mbf = ManualBeamFinder()
+
+        spinset = SpinSet(
+            "PLP0051296.nx.hdf",
+            "PLP0051294.nx.hdf",
+            up_down="PLP0051295.nx.hdf",
+            down_up="PLP0051297.nx.hdf",
+            data_folder=data_dir
+        )
+        spinset.du_opts.update({"manual_beam_find" : mbf, peak_pos : -1})
+        spinset.process_beams()
+
+        Parameters
+        ----------
+        reduction_options : dict
+            A single dict of options used to process all spectra. If
+            this is None, then process_beams will use individual dicts
+            for each spin channel
+        """
+
+        if reduction_options:
+            print(
+                "Applying the supplied reduction_options to all spin channels"
+            )
+            self.dd_opts = reduction_options.copy()
+            self.du_opts = reduction_options.copy()
+            self.ud_opts = reduction_options.copy()
+            self.uu_opts = reduction_options.copy()
+
+        for opts, beam in zip(
+            [self.dd_opts, self.du_opts, self.ud_opts, self.uu_opts],
+            [self.dd, self.du, self.ud, self.uu],
+        ):
```

Comment:
check for options that must be the same. It'll be required that the wavelength bins are the same.

Reply:
and add to documentation which attributes must be the same.